ape_hardhat/provider.py

```diff
@@ -320,6 +320,15 @@ def build_command(self) -> List[str]:
     def set_block_gas_limit(self, gas_limit: int) -> bool:
         return self._make_request("evm_setBlockGasLimit", [hex(gas_limit)]) is True
 
+    def set_code(self, address: AddressType, code: Union[str, bytes, HexBytes]) -> bool:
```

Comment:
As a future: we may want to add a custom type `ape.types`:

```python
ContractCode = Union[str, bytes, HexBytes]
```
and have that be the return type for `get_code()` and the param type here for code in `set_code()`

Reply:
should we make a new PR for this?

Reply:
It will be on the Ape core PR, not this one, this PR is good to go